app/src/sharedTest/java/org/oppia/android/app/databinding/MarginBindableAdaptersTest.kt

```diff
@@ -0,0 +1,328 @@
+      setLayoutMarginStartAndMarginEnd(textView)
+      textView.layoutDirection = View.LAYOUT_DIRECTION_RTL
+      return@runWithActivity textView
+    }
+    assertThat(textView.marginEnd.toFloat()).isWithin(TOLERANCE).of(
+      getContentMarginStart()
+    )
+    assertThat(textView.marginStart.toFloat()).isWithin(TOLERANCE).of(
+      getContentMarginEnd()
+    )
+  }
+
+  @Config(qualifiers = "land")
+  @Test
+  fun testMarginBindableAdapters_rtlIsEnabled_landscape_marginStartAndMarginEndForRtlIsCorrect() {
+    val textView = activityRule.scenario.runWithActivity {
+      val textView: TextView = it.findViewById(R.id.test_margin_text_view)
+      setLayoutMarginStartAndMarginEnd(textView)
+      textView.layoutDirection = View.LAYOUT_DIRECTION_RTL
+      return@runWithActivity textView
+    }
+    assertThat(textView.marginEnd.toFloat()).isWithin(TOLERANCE).of(
+      getContentMarginStart()
+    )
+    assertThat(textView.marginStart.toFloat()).isWithin(TOLERANCE).of(
+      getContentMarginEnd()
+    )
+  }
+
+  @Config(qualifiers = "sw600dp-port")
+  @Test
+  fun testMarginBindableAdapters_rtlIsEnabled_port_tablet_marginStartAndMarginEndForRtlIsCorrect() {
+    val textView = activityRule.scenario.runWithActivity {
+      val textView: TextView = it.findViewById(R.id.test_margin_text_view)
+      setLayoutMarginStartAndMarginEnd(textView)
+      textView.layoutDirection = View.LAYOUT_DIRECTION_RTL
+      return@runWithActivity textView
+    }
+    assertThat(textView.marginEnd.toFloat()).isWithin(TOLERANCE).of(
+      getContentMarginStart()
+    )
+    assertThat(textView.marginStart.toFloat()).isWithin(TOLERANCE).of(
+      getContentMarginEnd()
+    )
+  }
+
+  @Config(qualifiers = "sw600dp-land")
+  @Test
+  fun testMarginBindableAdapters_rtlEnabled_landscape_tablet_marginStartAndEndForRtlIsCorrect() {
+    val textView = activityRule.scenario.runWithActivity {
+      val textView: TextView = it.findViewById(R.id.test_margin_text_view)
+      setLayoutMarginStartAndMarginEnd(textView)
+      textView.layoutDirection = View.LAYOUT_DIRECTION_RTL
+      return@runWithActivity textView
+    }
+    assertThat(textView.marginEnd.toFloat()).isWithin(TOLERANCE).of(
+      getContentMarginStart()
+    )
+    assertThat(textView.marginStart.toFloat()).isWithin(TOLERANCE).of(
+      getContentMarginEnd()
+    )
+  }
+
+  private fun setLayoutMarginStartAndMarginEnd(textView: TextView) {
+    setLayoutMarginStart(
+      textView,
+      getContentMarginStart()
+    )
+    setLayoutMarginEnd(
+      textView,
+      getContentMarginEnd()
+    )
+  }
+
+  private fun getContentMarginEnd(): Float {
+    return context.resources.getDimension(R.dimen.content_item_exploration_view_margin_end)
+  }
+
+  private fun getContentMarginStart(): Float {
+    return context.resources.getDimension(R.dimen.content_item_exploration_view_margin_start)
+  }
+
+  private inline fun <reified V, A : Activity> ActivityScenario<A>.runWithActivity(
```

Comment:
Is it ok to generalize this function, so that we can use it from a central utility wherever it required and we can its own test file too.

Reply:
May be we can do that, we can have a separate PR to generalize this function otherwise.

Reply:
Sounds good Let's create an issue for this.